sphinxgallery/backreferences.py

```diff
@@ -0,0 +1,167 @@
+            local_name = name.split('.', 1)[0]
+            remainder = name[len(local_name):]
+            if local_name in self.imported_names:
+                # Join import path to relative path
+                full_name = self.imported_names[local_name] + remainder
+                yield name, full_name
+
+
+def get_short_module_name(module_name, obj_name):
+    """ Get the shortest possible module name """
+    parts = module_name.split('.')
+    short_name = module_name
+    for i in range(len(parts) - 1, 0, -1):
+        short_name = '.'.join(parts[:i])
+        try:
+            exec('from %s import %s' % (short_name, obj_name))
+        except ImportError:
+            # get the last working module name
+            short_name = '.'.join(parts[:(i + 1)])
+            break
+    return short_name
+
+
+def identify_names(code):
+    """Builds a codeobj summary by identifying and resovles used names
+
+    >>> code = '''
+    ... from a.b import c
+    ... import d as e
+    ... print(c)
+    ... e.HelloWorld().f.g
+    ... '''
+    >>> for name, o in sorted(identify_names(code).items()):
+    ...     print(name, o['name'], o['module'], o['module_short'])
+    c c a.b a.b
+    e.HelloWorld HelloWorld d d
+    """
+    finder = NameFinder()
+    finder.visit(ast.parse(code))
+
+    example_code_obj = {}
+    for name, full_name in finder.get_mapping():
+        # name is as written in file (e.g. np.asarray)
+        # full_name includes resolved import path (e.g. numpy.asarray)
+        module, attribute = full_name.rsplit('.', 1)
+        # get shortened module name
+        module_short = get_short_module_name(module, attribute)
+        cobj = {'name': attribute, 'module': module,
+                'module_short': module_short}
+        example_code_obj[name] = cobj
+    return example_code_obj
+
+
+def scan_used_functions(example_file, gallery_conf):
+    """save variables so we can later add links to the documentation"""
+    example_code_obj = identify_names(open(example_file).read())
+    if example_code_obj:
+        codeobj_fname = example_file[:-3] + '_codeobj.pickle'
+        with open(codeobj_fname, 'wb') as fid:
+            pickle.dump(example_code_obj, fid, pickle.HIGHEST_PROTOCOL)
+
+    backrefs = set('{module_short}.{name}'.format(**entry)
+                   for entry in example_code_obj.values()
+                   if entry['module'].startswith(gallery_conf['doc_module']))
+
+    return backrefs
+
+
+def _thumbnail_div(full_dir, fname, snippet):
+    """Generates RST to place a thumbnail in a gallery"""
+    thumb = os.path.join(full_dir, 'images', 'thumb',
+                         'sphx_glr_%s.png' % fname[:-3])
+    link_name = os.path.join(full_dir, fname)
+    ref_name = link_name.replace(os.path.sep, '_')
+    if ref_name.startswith('._'):
+        ref_name = ref_name[2:]
+
+    out = """
+.. raw:: html
+
+    <div class="sphx-glr-thumbContainer" tooltip="{}">
+
+.. figure:: /{}
+    :target: /{}.html
```

Comment:
Yes, I also agree on that. Although once published online I doesn't do much of a difference. The thing I fear is that it will require to extended sphinx to accept the :doc: or :ref: directive inside the figure environment. at target can only take urls or same page references. And I'm still trying to figure out how does sphinx work at this level.